backend/src/graphql/server.ts

```diff
@@ -0,0 +1,56 @@
+import { ApolloServer } from '@apollo/server';
+import { resolvers } from './resolvers';
+import {
+  constraintDirective,
+  constraintDirectiveTypeDefs
+} from 'graphql-constraint-directive';
+import DataLoader from 'dataloader';
+import { authorService } from 'src/services/author';
+import { bookService } from 'src/services/book';
+import { makeExecutableSchema } from '@graphql-tools/schema';
+import typeDefs from './schema.graphql';
+
+let schema = makeExecutableSchema({
+  typeDefs: [constraintDirectiveTypeDefs, typeDefs],
+  resolvers
+});
+schema = constraintDirective()(schema);
+
+const gqlServer = new ApolloServer({
+  schema
+});
+
+const getDataLoaders = () => {
+  const authorLoader = new DataLoader(async (ids: readonly string[]) => {
+    authorLoader.clearAll(); // clear out cache for fetching the most updated data from the source
```

Comment:
If we are clearing the cache every time a request is made to `DataLoader`, will we ever hit the cache?

Reply:
Yes, we do. This pattern ensures that data loaders are cleared for every request to be updated for every call in that request.

Reply:
Have the same question as @irshadSprinto. We seem to be clearing on each call to Dataloader, not once per request. :thinking: